### DISCUSSION TITLE
Incremental training within rolling window

### DISCUSSION THREAD

**Discussion by occoder** [online-ml/river#479] [Q&A] [answered]
It's certain that my model does drift as time passes by in nature.
Can I define a certain rolling window that could make the model sink in latest data meanwhile forgetting oldest data?
In other words, keep the dataset used for training model act like FIFO.
Is there any best practice to reference?
Thanks

**Comment by MaxHalford** [comment, marked as answer]
Hello there. In general you can't explicitely set a window. It all depends on the model you're using. For instance, with `neighbors.KNNClassifier`, there's a `window_size` parameter. But with `linear_model.LogisticRegression` there's no such parameter. However, you can increase/lower the learning rate of the optimizer to produce the same effect. I don't think there's any best practice: it depends on your data. You should simply do some progressive validation with different parameters and see what happens.

I hope this helps!

**Reply by occoder** [reply]
Hi Max,
Thanks for your advice.
Batching training seems to favor learning rate **decay** over time...
But my application mostly values the latest data. I think this is often the case for many streaming scenarios.
As such, a straightforward explicit rolling window setting might be a unique feature to incremental machine learning package.
Learning from the latest data meanwhile unlearning off the oldest data could be a very attractive capability.
Would you like to share more insight on this?
Thanks again

**Reply by MaxHalford** [reply]
That's what I'm trying to say: using a constant learning rate will result in a model that forgets older data and favours new data. Of course the forgetting process isn't explicit, as with k-nearest neighbors, but it works.
